internal/rule/unix_rules.go

```diff
@@ -0,0 +1,70 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build !windows
+
+package rule
+
+import (
+	"github.com/aws/amazon-cloudwatch-agent-test/filesystem"
+	"log"
+)
+
+type FilePermissionExpected struct {
+	PermissionCompared filesystem.FilePermission
+	ShouldExist        bool
+}
```

Comment:
To make it easier for developers to tell which interface(s) you intend to implement it helps to add a line like:
```
// Verify Foo implements Bar
var _ MyPkg.Bar = (*Foo)(nil)
```